cheroot/server.py

```diff
@@ -654,7 +656,10 @@ def read_request_headers(self):
 
         # then all the http headers
         try:
-            read_headers(self.rfile, self.inheaders)
+            # Configurable using "`server.drop_underscore_headers": True`:
+#             drop_underscores = self.server.server_adapter.drop_underscore_headers
```

Comment:
No, I left this comment for discussion, since I noticed that this code
```py
    cherrypy.config.update({"server.drop_underscore_headers": True})
    cherrypy.quickstart(HelloWorld())
```
modified the attribute on `self.server.server_adapter.drop_underscore_headers` (not `self.server.server_adapter.drop_underscore_headers`).
So I am not sure if we should declare this option on the adapter instead of the server object.